symbolic-symcache/src/lib.rs

```diff
@@ -303,3 +306,67 @@ impl<'slf, 'd: 'slf> AsSelf<'slf> for SymCache<'d> {
         self
     }
 }
+
+/// Appends the provided path to the provided buffer, normalizing it in the process.
+///
+/// "Normalizing" in this context means:
+/// * if the path is absolute, the buffer will be cleared prior to appending.
+/// * `.` components in the path will be skipped.
+/// * `..` components in the path will delete the previous component from the buffer.
+pub(crate) fn normalize_path(buffer: &mut Utf8PathBuf, path: &Utf8Path) {
```

Comment:
I would rather call this `append_path_normalized`. Just going by the name, I would assume that it normalizes the `&mut self` parameter.